Tests/MongoSwiftTests/UnifiedTestRunner/Context.swift

```diff
@@ -19,10 +21,25 @@ class Context {
         self.internalClient = internalClient
     }
 
-    /// Executes a closure with the given path element added to the path, removing it after the closure is complete.
+    /// Method to manually close failpoints given the new `async` implementation
+    func disableFailpoints() async {
+        for failpointGuard in self.enabledFailPoints {
+            await failpointGuard.failPoint.disable(using: self.internalClient.anyClient)
+        }
+    }
+
+    /// Executes a closure with the given path element added to the path, removing it after closure completes.
     func withPushedElt<T>(_ elt: String, work: () throws -> T) rethrows -> T {
         self.path.append(elt)
         defer { self.path.removeLast() }
         return try work()
     }
+
+    /// Executes an async closure with the given path element added to the path, removing it after closure completes.
+    func withPushedElt<T>(_ elt: String, work: () async throws -> T) async rethrows -> T {
```

Comment:
aside: in the future with [`reasync`](https://github.com/apple/swift-evolution/blob/main/proposals/0296-async-await.md#reasync), we would only need to have one version of this method -- it would allow us to say that the method is only `async` if the provided closure is `async`. similar to how `rethrows` lets us say it only `throws` if the provided closure `throws`